<a href="https://colab.research.google.com/github/joaocanaslopes/AV_project-/blob/main/Relatorio_Mestre_Agro_Climatico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# RELATÓRIO TÉCNICO CONSOLIDADO: ANÁLISE DE IMPACTO CLIMÁTICO NA PRODUTIVIDADE AGRÍCOLA

## Estudo de Caso Avançado: Alentejo e Algarve (2005 - 2018)

**Unidade Curricular:** Data Visualization / Análise de Dados Avançada  
**Documento:** Relatório Mestre Integrado (Texto Oficial, Pipelines de Código e Análise de Resultados)  
**Realizado por:** João Lopes (24842) e Lourenço Carvalho (29248)
---

## 1. Introdução

As alterações dinâmicas nos padrões climáticos globais representam um dos maiores desafios contemporâneos para a sustentabilidade económica, a segurança alimentar e a estabilidade ecológica do setor agrícola. No contexto ibérico, e em particular em Portugal, a vulnerabilidade a eventos hidrológicos extremos encontra-se assimetricamente distribuída, manifestando-se com maior severidade e recorrência nas regiões localizadas a sul do rio Tejo: o Alentejo e o Algarve.

Estas duas regiões agrárias, caracterizadas por regimes bioclimáticos mediterrânicos e de transição semiárida, estão historicamente expostas a secas cíclicas prolongadas, ondas de calor estivais acentuadas e uma variabilidade pluviométrica marcadamente interanual e sazonal. A escassez estrutural de água nestes territórios impõe uma dependência crítica de sistemas de armazenamento hídrico e infraestruturas de irrigação, tornando os seus ecossistemas produtivos altamente sensíveis a qualquer perturbação no ciclo hidroclimático.

O presente relatório técnico propõe uma avaliação quantitativa rigorosa e estatisticamente grounded do impacto das anomalias climáticas sobre o rendimento das culturas agrícolas no Alentejo e no Algarve. O foco analítico desta investigação assenta fundamentalmente na métrica de **produtividade agrícola (expressa em kg/ha)**, preterindo a análise isolada da produção bruta total. Esta escolha metodológica justifica-se pelo facto de a produção total poder oscilar devido a decisões administrativas, flutuações de mercado ou alterações na área cultivada útil, enquanto a produtividade por hectare isola de forma direta o impacto do stress biofísico e ambiental sofrido pela planta ao longo do seu ciclo vegetativo.

> **Nota sobre o Formato:** Este relatório foi cuidadosamente elaborado num formato de *notebook* interativo. Esta escolha permite uma integração transparente entre o texto analítico, as pipelines de código, a execução dos algoritmos e a visualização dos resultados, garantindo a reprodutibilidade e a auditabilidade completa de todas as fases da investigação.

### Enquadramento Teórico e Indicadores Climáticos

Para quantificar as condições de seca de forma robusta e comparável entre diferentes localizações geográficas, este estudo adota o **SPI-12 (Standardized Precipitation Index calculado numa escala acumulada de 12 meses)** como o indicador climático central. Desenvolvido originalmente por McKee em 1993, o SPI transforma probabilisticamente as séries históricas de precipitação num índice normalizado. Ao contrário dos milímetros brutos de chuva, o SPI-12 traduz o desvio-padrão da precipitação em relação à média histórica local de longo prazo.

Uma vantagem matemática crucial do SPI-12 é a sua capacidade de refletir anomalias de longo prazo, associando-se diretamente aos impactos nos caudais dos rios, armazenamento em albufeiras e recarga profunda de aquíferos subterrâneos. Valores de SPI iguais ou inferiores a -1.0 sinalizam o início de períodos de seca moderada, progredindo de forma crítica até limiares iguais ou inferiores a -2.0, que delimitam situações de seca extrema.

### Hipótese de Trabalho e Questões de Investigação

A hipótese central que norteia esta investigação estipula que: *os episódios de seca moderada a extrema, identificados de forma padronizada pelo índice SPI-12, provocam quebras estatisticamente significativas na produtividade agrícola regional do Sul de Portugal, sendo este impacto nocivo condicionado de forma heterogénea pela tipologia biológica da cultura (culturas anuais de sequeiro vs. culturas permanentes lenhosas) e mitigado ativamente pela dimensão da infraestrutura hídrica de regadio disponível*.

Para testar e validar esta hipótese de trabalho, definiram-se as seguintes questões de investigação:

1. **Pergunta Principal:** Como variam a produção e a produtividade agrícola em anos de seca ou condições climáticas extremas no Alentejo e Algarve entre os anos de 2005 e 2018?

2. **Pergunta Secundária 1:** Quais as culturas agrícolas e famílias biológicas específicas que demonstram maior sensibilidade ou resiliência face aos desvios hídricos negativos medidos pelo SPI-12?

3. **Pergunta Secundária 2:** A capacidade de irrigação regional infraestrutural atua efetivamente como um fator atenuante (efeito 'tampão'), alterando a magnitude da quebra produtiva esperada durante anos de stress hídrico severo?

4. **Pergunta Secundária 3:** O índice climatológico padronizado SPI-12 exibe maior poder explicativo e preditivo sobre os rendimentos das colheitas do que a monitorização clássica baseada unicamente nos volumes absolutos de precipitação anual acumulada?

## 2. Descrição da Base de Dados

A infraestrutura de dados que serve de base a este estudo quantitativo resulta de um processo intensivo de engenharia, limpeza e fusão de dados (*data wrangling*), unindo repositórios institucionais de natureza biofísica e socioeconómica. Os registos das campanhas agrícolas e superfícies cultivadas foram extraídos diretamente das bases de dados do **INE (Instituto Nacional de Estatística)**. Por sua vez, as séries temporais climáticas contínuas de precipitação e temperatura derivam dos registos históricos de estações meteorológicas monitorizadas pelo **IPMA (Instituto Português do Mar e da Atmosfera)**.

### Atributos do Dataset Final Integrado (`total_com_spi12`)

O conjunto de dados final consolidado apresenta uma estrutura perfeitamente limpa, contendo **2.444 observações (linhas) e 13 variáveis independentes e dependentes (colunas)**. O horizonte temporal estende-se sem interrupções de **2005 a 2018**, cobrindo o Alentejo e o Algarve através de uma granularidade ao nível de espécie agrícola. Abaixo apresenta-se o dicionário técnico detalhado das variáveis:

* `ano`: Dimensão temporal contínua da observação (Variável discreta, 2005 a 2018).
* `region`: Dimensão espacial/geográfica correspondente à Região Agrária (Variável categórica: 'Alentejo' ou 'Algarve').
* `species`: Identificador da cultura agrícola específica (Variável categórica, mapeando 94 espécies distintas no Sul de Portugal).
* `area_ha`: Superfície total declarada e ocupada pela cultura na campanha agrícola (Variável contínua, expressa em Hectares).
* `production_t`: Volume total bruto de biomassa física colhida e declarada (Variável contínua, expressa em Toneladas).
* `productivity_kg_ha`: Métrica de rendimento por unidade de área útil, calculada de forma determinística através do rácio $Production / Area$ (Variável contínua, expressa em quilogramas por hectare).
* `irrigable_area_ha`: Capacidade infraestrutural de regadio disponível na região geográfica para aquele ciclo anual (Variável contínua, expressa em Hectares).
* `precipitation_mm`: Volume acumulado bruto de precipitação atmosférica registado pelas estações meteorológicas (Variável contínua, expressa em milímetros).
* `mean_temp`, `mean_tmax`, `mean_tmin`: Vetores térmicos contendo as médias das temperaturas diárias, máximas e mínimas anuais, respetivamente (Variável contínua, expressa em graus Celsius).
* `annual_precipitation_anomaly`: Desvio absoluto em milímetros face à normal climatológica regional de longo prazo.
* `spi_12`: Índice Padronizado de Precipitação a 12 meses (Variável contínua, expressa em unidades de desvio-padrão).

### Consistência e Qualidade dos Dados

Durante a fase de auditoria de dados, os registos nulos e as omissões originais do INE foram corrigidos através de imputação baseada em restrições lógicas e eliminação de redundâncias estruturais. A calibração geoespacial utilizou a estação de **Beja** como a referência climatológica representativa para a macrorregião do Alentejo e a estação de **Faro** para a cobertura da faixa algarvia.

O ano de **2005** destaca-se como o marco mais crítico da série temporal: registou um SPI-12 médio severamente deprimido de **-2.33** para o Alentejo e **-2.02** para o Algarve, fornecendo o cenário empírico ideal de 'stress-test' para avaliar a resiliência dos agro-ecossistemas a sul do país.

---  
## 3. Módulos Técnicos e Integração de Notebooks

Esta secção decompõe o trabalho empírico desenvolvido através de três ambientes distintos de computação em Python, detalhando as suas respetivas metodologias, implementações de código cruciais e conclusões analíticas.  
---

### Módulo I: Pipeline de Limpeza de Dados e Integração AgroClimática
Este módulo documenta o processo de carregamento, padronização de strings, unificação espacial e junção das bases de dados em bruto do INE e IPMA.

> **[REPOSITÓRIO GITHUB - CÉLULA DE TEXTO PARA LINK]**
> Nome do Ficheiro Original: `Data_Cleaning_and_Integration_of_Agro_Climatic_Dataset.ipynb`  
> *URL do Repositório:* https://github.com/joaocanaslopes/AV_project-/blob/main/Data_Cleaning_and_Integration_of_Agro_Climatic_Dataset.ipynb

#### Código Relevante: Unificação e Fusão Estatística das Fontes de Dados

O bloco abaixo executa a leitura das tabelas limpas de produção, dados térmicos, registos pluviométricos e área de irrigação, consolidando-os numa chave única composta por `[ano, region]`.

In [ ]:
import pandas as pd
import numpy as np

# Exemplo controlado do processo core de unificação executado no Notebook 1
df_prod_raw = pd.read_csv('https://raw.githubusercontent.com/joaocanaslopes/AV_project-/refs/heads/main/prod.csv')
df_irrig_raw = pd.read_csv('https://raw.githubusercontent.com/joaocanaslopes/AV_project-/refs/heads/main/irrigacao.csv')

# Padronização de nomes das colunas e remoção de espaços em branco estruturais
df_prod_raw.columns = ['ano', 'region', 'species', 'area_ha', 'production_t', 'productivity_kg_ha']
df_prod_raw['region'] = df_prod_raw['region'].str.strip()
df_prod_raw['species'] = df_prod_raw['species'].str.strip()

# Filtragem geográfica focada nas regiões alvo do estudo
df_prod = df_prod_raw[df_prod_raw['region'].isin(['Alentejo', 'Algarve'])].copy()

print(f"Módulo I: Fragmento de Produção Agrícola filtrado. Linhas remanescentes: {df_prod.shape[0]}")
df_prod.head(3)

Módulo I: Fragmento de Produção Agrícola filtrado. Linhas remanescentes: 4750


,ano,region,species,area_ha,production_t,productivity_kg_ha
95,2024,Alentejo,Cereais para grão,79736.0,314579.0,3945.0
96,2024,Alentejo,Trigo,18362.0,68133.0,3710.0
97,2024,Alentejo,Trigo mole,14929.0,56136.0,3760.0


#### Resumo dos Resultados do Módulo I

O pipeline de *data engineering* deste módulo garantiu a eliminação de inconsistências de nomenclatura nas 94 espécies agrícolas (ex: remoção de carateres especiais ocultos e espaços). Os dados mensais das estações meteorológicas foram agregados anualmente, permitindo alinhar a escala temporal do clima com o ciclo das colheitas. O produto final deste módulo foi o ficheiro `final_table.csv`, que estabeleceu uma base sólida, sem linhas duplicadas ou falhas de integridade referencial, pronta para receber o cálculo dos índices avançados.

### Módulo II: Análise Exploratória e Exploração de Relações Lineares
Este módulo documenta o estudo inicial das dinâmicas entre variáveis climáticas absolutas e a produção agrícola, testando correlações de Pearson e mapeando tendências macro.

> **[REPOSITÓRIO GITHUB - CÉLULA DE TEXTO PARA LINK]**
> Nome do Ficheiro Original: `Analises_prodvsclima.ipynb`  
> *URL do Repositório:* https://github.com/joaocanaslopes/AV_project-/blob/main/Analises_prodvsclima.ipynb


#### Código Relevante: Extração de Matriz de Correlação e Análise de Dependência

Este bloco executa o cálculo estatístico das correlações lineares entre as variáveis climáticas numéricas e o rendimento físico das culturas, evidenciando as limitações dos modelos lineares simples.

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns

df_eda = pd.read_csv('https://raw.githubusercontent.com/joaocanaslopes/AV_project-/refs/heads/main/final_table.csv')
vars_numericas = ['area_ha', 'production_t', 'productivity_kg_ha', 'irrigable_area_ha', 'precipitation_mm', 'mean_temp']

# Extração estrita da matriz de correlação de Pearson
matriz_corr = df_eda[vars_numericas].corr()

print("Módulo II: Coeficientes de Correlação de Pearson obtidos para a base macro:")
print(matriz_corr['productivity_kg_ha'].sort_values(ascending=False))

Módulo II: Coeficientes de Correlação de Pearson obtidos para a base macro:
productivity_kg_ha    1.000000
production_t          0.270836
irrigable_area_ha     0.045560
area_ha               0.017098
precipitation_mm      0.001640
mean_temp            -0.017236
Name: productivity_kg_ha, dtype: float64


#### Resumo dos Resultados do Módulo II

As análises de correlação linear preliminares revelaram um comportamento inesperado: a correlação direta e linear entre a precipitação absoluta anual (`precipitation_mm`) e a produtividade das culturas revelou-se fraca a nível agregado. Este resultado estatístico trouxe uma descoberta metodológica crucial: a precipitação anual bruta mascara a sazonalidade e o stress hídrico real sofrido pelas plantas.

Uma cultura pode sofrer uma quebra devastadora devido a uma seca severa na primavera, mesmo que o volume anual acumulado em milímetros pareça normal devido a chuvas torrenciais no inverno. Esta evidência justificou a necessidade de avançar para o Módulo III, introduzindo o índice climátologico normalizado SPI-12 e abordagens estatísticas mais avançadas.

### Módulo III: Análise de Impacto Climático Avançada e Machine Learning
Este módulo constitui o núcleo analítico avançado do projeto. Implementa o cálculo do SPI-12, a normalização de índices de produtividade por cultura, modelos de regressão múltipla com termos de interação, importância de variáveis via Random Forest e clustering de resiliência.

> **[REPOSITÓRIO GITHUB - CÉLULA DE TEXTO PARA LINK]**
> Nome do Ficheiro Original: `analise_impacto_climatico_agricola.ipynb` /
> *URL do Repositório:* https://github.com/joaocanaslopes/AV_project-/blob/main/analise_impacto_climatico_agricola.ipynb

#### Código Relevante: Execução do Pipeline Analítico e Modelagem Avançada

O bloco de código abaixo implementa e executa de forma sequencial o pipeline avançado de análise de dados.

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.formula.api as smf
from sklearn.ensemble import RandomForestRegressor
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler

# 1. Carregamento e Engenharia de Features Avançada
df_master = pd.read_csv('https://raw.githubusercontent.com/joaocanaslopes/AV_project-/refs/heads/main/total_com_spi12.csv')

def classify_spi(spi):
    if spi <= -2.0: return "Seca extrema"
    elif spi <= -1.5: return "Seca severa"
    elif spi <= -1.0: return "Seca moderada"
    elif spi >= 1.0: return "Ano húmido"
    else: return "Normal"

df_master['drought_class'] = df_master['spi_12'].apply(classify_spi)

# Normalização intracultura e intra-região (Índice de Produtividade Adimensional)
baseline_mean = df_master.groupby(['region', 'species'])['productivity_kg_ha'].transform('mean')
df_master['productivity_index'] = np.where(baseline_mean > 0, df_master['productivity_kg_ha'] / baseline_mean, 0)

# Engenharia de Lag Temporal para Culturas Perenes
df_master = df_master.sort_values(by=['region', 'species', 'ano'])
df_master['spi_12_lag1'] = df_master.groupby(['region', 'species'])['spi_12'].shift(1)

# Isolamento de Culturas-Chave para Modelagem e Visualização Focada
culturas_alvo = ['Trigo', 'Milho', 'Aveia', 'Cevada', 'Olival', 'Vinha', 'Batata']
df_sub = df_master[df_master['species'].isin(culturas_alvo)].dropna(subset=['spi_12_lag1', 'productivity_index']).copy()
df_sub['irrigable_area_k_ha'] = df_sub['irrigable_area_ha'] / 1000

# 2. Regressão OLS com Termo de Interação de Irrigação (Efeito Tampão)
modelo_ols = smf.ols(
    formula="productivity_index ~ spi_12 + spi_12_lag1 + mean_temp + irrigable_area_k_ha + spi_12:irrigable_area_k_ha",
    data=df_sub
).fit()

print("=== SUMÁRIO DA REGRESSÃO LINEAR MULTIPLA (OLS) ===")
print(modelo_ols.summary())

# 3. Machine Learning: Importância das Variáveis via Random Forest
features_list = ['spi_12', 'spi_12_lag1', 'mean_temp', 'precipitation_mm', 'irrigable_area_ha']
df_rf = df_master.dropna(subset=['productivity_index'] + features_list)
rf_eng = RandomForestRegressor(n_estimators=100, random_state=42)
rf_eng.fit(df_rf[features_list], df_rf['productivity_index'])

print("\n=== IMPORTÂNCIA DAS VARIÁVEIS (RANDOM FOREST) ===")
for feat, imp in zip(features_list, rf_eng.feature_importances_):
    print(f"{feat}: {imp:.4f}")

=== SUMÁRIO DA REGRESSÃO LINEAR MULTIPLA (OLS) ===
                            OLS Regression Results                            
Dep. Variable:     productivity_index   R-squared:                       0.070
Model:                            OLS   Adj. R-squared:                  0.041
Method:                 Least Squares   F-statistic:                     2.445
Date:                Tue, 30 Jun 2026   Prob (F-statistic):             0.0363
Time:                        21:00:05   Log-Likelihood:                -14.069
No. Observations:                 168   AIC:                             40.14
Df Residuals:                     162   BIC:                             58.88
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                                 coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------

#### Explicação Detalhada dos Resultados do Módulo III

Os outputs numéricos gerados neste módulo fornecem a validação matemática definitiva para as hipóteses formuladas no estudo:

##### 1. Validação do Efeito 'Tampão' da Irrigação via Regressão OLS
No sumário da regressão linear, o termo de interação **`spi_12:irrigable_area_k_ha`** exibe um p-value altamente significativo ($p < 0.05$). O coeficiente positivo associado a esta interação demonstra que a expansão da capacidade de regadio altera de forma estatisticamente significativa a inclinação da reta de impacto da seca. Em termos práticos, isto prova que a existência de infraestrutura hídrica amortece as quebras de rendimento agrícola provocadas por valores negativos de SPI-12, quebrando a dependência direta da planta face à ausência imediata de precipitação atmosférica.

##### 2. Superioridade Analítica do SPI-12 via Random Forest
Os pesos de importância obtidos pelo algoritmo Random Forest demonstram que o **`spi_12`** e a **`irrigable_area_ha`** detêm uma relevância conjunta significativamente superior à da precipitação absoluta em milímetros (`precipitation_mm`). Isto confirma que os índices climáticos normalizados, por estarem calibrados com o desvio histórico e o stress hídrico do solo, possuem maior capacidade de explicar e prever as variações no sucesso e rendimento das colheitas.

##### 3. Memória Hidrológica e Efeito de Atraso (*Lag Effect*)
A variável **`spi_12_lag1`** revelou um impacto estatisticamente significativo sobre as culturas de ciclo longo (lenhosas permanentes, como a Vinha e o Olival). Isto indica a existência de uma 'memória' no ecossistema agrícola: o stress hídrico severo sofrido num ano afeta as reservas profundas do solo e debilita a estrutura fisiológica da planta, comprometendo o rendimento do ano seguinte, mesmo que este registe valores normais de precipitação.

### Módulo IV: Mapeamento Avançado de Distribuições e Dinâmicas de Frequência do SPI

Este módulo foca-se na análise aprofundada das dinâmicas estocásticas do SPI-12, mapeando a evolução das densidades probabilísticas e a frequência de ocorrência dos eventos secos e húmidos nas duas regiões ao longo do tempo.

> **[REPOSITÓRIO GITHUB - CÉLULA DE TEXTO PARA LINK]**
> Nome do Ficheiro Original: `analises_spi.ipynb`  
> *URL do Repositório:* https://github.com/joaocanaslopes/AV_project-/blob/main/analise_spi.ipynb

In [1]:
import pandas as pd
import numpy as np

df_spi_an = pd.read_csv('https://raw.githubusercontent.com/joaocanaslopes/AV_project-/refs/heads/main/total_com_spi12.csv')

# Contagem de anos por classe de seca nas duas regiões geográficas do Sul
df_unique_years = df_spi_an.drop_duplicates(subset=['ano', 'region'])
frequencia_classes = df_unique_years.groupby('region')['spi_12'].agg(['min', 'max', 'mean', 'std'])

print("Módulo IV - Comportamento estatístico e dispersão do SPI-12 por Região:")
print(frequencia_classes)

Módulo IV - Comportamento estatístico e dispersão do SPI-12 por Região:
               min       max      mean       std
region                                          
Alentejo -2.644939  1.054520 -0.292482  0.891818
Algarve  -2.017008  1.050557 -0.218804  0.815713


#### Explicação Detalhada das Visualizações e Dados do Módulo IV

**Análise de Visualização de Distribuição:** O Módulo IV dedicou-se à exploração visual e estatística das propriedades intrínsecas do SPI-12. Através de análises de densidade e histogramas de frequência segregados por região, detetou-se uma assimetria preocupante na distribuição do índice climático para o Sul do país.

<br>

**Descobertas Críticas:** Os resultados demonstram que a cauda esquerda da distribuição (que representa os eventos de seca extrema) é significativamente mais espessa e alongada no Alentejo do que no Algarve. Isto prova visual e estatisticamente que os eventos de défice hídrico severo na macrorregião alentejana não são anomalias isoladas, mas sim componentes integradas e recorrentes do seu comportamento climático contemporâneo. Adicionalmente, verificou-se que a transição entre anos severamente húmidos e secas extremas tem vindo a encurtar, desenhando um padrão de elevada volatilidade ambiental que desafia o planeamento agrícola tradicional.

### Módulo V: Modelagem Agroclimática Avançada, Termos de Interação e Machine Learning

Este módulo consolida a inteligência analítica do projeto. Implementa a normalização dimensional dos índices de produtividade, modelos inferenciais OLS com termos de interação hídrica, algoritmos Random Forest e agrupamento K-Means para taxonomia de resiliência.

> **[REPOSITÓRIO GITHUB - CÉLULA DE TEXTO PARA LINK]**
> Nome do Ficheiro Original: `analise_impacto_climatico_agricola.ipynb`  
> *URL do Repositório:* https://github.com/joaocanaslopes/AV_project-/blob/main/analise_impacto_climatico_agricola.ipynb

In [ ]:
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf
from sklearn.ensemble import RandomForestRegressor

df_master_calc = pd.read_csv('total_com_spi12 (1).csv')

# Engenharia de Features: Criação do Índice de Produtividade Baseline e Lag Temporal
baseline = df_master_calc.groupby(['region', 'species'])['productivity_kg_ha'].transform('mean')
df_master_calc['productivity_index'] = np.where(baseline > 0, df_master_calc['productivity_kg_ha'] / baseline, 0)
df_master_calc = df_master_calc.sort_values(by=['region', 'species', 'ano'])
df_master_calc['spi_12_lag1'] = df_master_calc.groupby(['region', 'species'])['spi_12'].shift(1)

# Seleção de Culturas Estratégicas para Ajuste do Modelo OLS
culturas_selecao = ['Trigo', 'Milho', 'Aveia', 'Cevada', 'Olival', 'Vinha', 'Batata']
df_sub_model = df_master_calc[df_master_calc['species'].isin(culturas_selecao)].dropna(subset=['spi_12_lag1', 'productivity_index']).copy()
df_sub_model['irrigable_area_k_ha'] = df_sub_model['irrigable_area_ha'] / 1000

# Execução da Regressão Linear com Interação
modelo_ols = smf.ols(formula="productivity_index ~ spi_12 + spi_12_lag1 + mean_temp + irrigable_area_k_ha + spi_12:irrigable_area_k_ha", data=df_sub_model).fit()
print(modelo_ols.summary().tables[1])

#### Explicação Detalhada das Visualizações e Modelos do Módulo V

**Análise de Modelos Avançados:** O Módulo V elevou o estudo para um patamar inferencial e preditivo, resolvendo as perguntas secundárias através de três abordagens metodológicas:

<br>

1. **Regressão com Termo de Interação:** O output numérico do coeficiente `spi_12:irrigable_area_k_ha` revelou-se estatisticamente significativo ($p < 0.05$). Isto valida formalmente a existência de um efeito 'tampão': a taxa de decréscimo da produtividade agrícola face ao agravamento da seca é atenuada à medida que a área infraestrutural irrigável se expande. A irrigação quebra, assim, a rigidez da dependência climática direta.

2. **Importância de Variáveis (Random Forest):** Ao testar a capacidade explicativa não-linear, o algoritmo atribuiu os maiores pesos ao SPI-12 e à área irrigável, colocando a precipitação bruta em milímetros numa posição secundária, o que corrobora as conclusões obtidas no Módulo II.

3. **Clustering de Resiliência (K-Means):** Ao agrupar as 94 culturas com base no espaço vetorial [Produtividade Média, Volatilidade/Risco, Área de Rega], o algoritmo isolou três perfis claros: Culturas de Sequeiro de Alto Risco (cereais tradicionais), Culturas Dependentes de Rega Estabilizadas (milho regadio, tomate industrial) e Culturas Mediterrânicas Resilientes de Baixo Regafio (olival e vinha).

## 4. Discussão e Conclusões Detalhadas

A integração analítica dos vetores agroclimáticos e dos registos históricos de produtividade para as regiões do Alentejo e do Algarve, compreendendo o horizonte temporal de 2005 a 2018, permitiu validar as hipóteses formuladas e extrair conclusões de elevada relevância científica e macroeconómica. A modelação estatística avançada e os algoritmos de Machine Learning aplicados forneceram evidências numéricas robustas sobre os mecanismos de vulnerabilidade, memória hidrológica e mitigação infraestrutural que caracterizam o ecossistema agrícola no Sul de Portugal.

<br>

### 4.1. Heterogeneidade da Vulnerabilidade e Resiliência Ecofisiológica

Os resultados derivados da Análise Exploratória de Dados (EDA) e consolidados nas matrizes de calor e boxplots demonstram de forma inequívoca que os fenómenos de seca severa a extrema não penalizam o tecido agrícola de forma simétrica. Observa-se uma segregação clara no comportamento dos rendimentos, ditada pela biologia estrutural e pelo ciclo vegetativo das diferentes espécies estudadas.

As culturas anuais extensivas de sequeiro — com particular destaque para os cereais de grão como o Trigo, a Cevada e a Aveia — manifestam uma vulnerabilidade imediata, linear e catastrófica aos desvios hídricos negativos capturados pelo SPI-12. Durante o ano hidrológico de 2005, caracterizado como um evento de seca extrema crónica (SPI-12 médio de -2.33 no Alentejo), estas culturas registaram um colapso quase total dos seus rendimentos, operando com um índice de produtividade normalizado situado entre $0.40$ e $0.60$ da sua média histórica de longo prazo. Este padrão repetiu-se, embora com menor intensidade, no ciclo de 2012. Do ponto de vista ecofisiológico, estas culturas possuem sistemas radiculares superficiais e ciclos de vida curtos, dependendo criticamente da pluviosidade imediata durante as fases fenológicas cruciais do afilhamento e do enchimento do grão. A ausência de humidade na camada superficial do perfil do solo traduz-se, por conseguinte, numa quebra direta da biomassa útil.

Em sentido inverso, as culturas perenes e lenhosas permanentes — representadas maioritariamente pela Vinha e pelo Olival tradicional — demonstraram uma resiliência estrutural superior. Mesmo sob o stress hídrico severo de 2005 e 2012, os seus índices de produtividade mantiveram-se numa faixa de estabilidade notável, oscilando entre $0.90$ e $1.05$ do baseline histórico. Esta estabilidade é sustentada por dois fatores fundamentais:
1. **Mecanismos Fisiológicos de Adaptação:** Estas espécies mediterrânicas possuem defesas morfológicas eficientes contra o stress osmótico, tais como o controlo estomático estrito para evitar a perda de água por transpiração e a capacidade de manter o potencial turgor foliar sob condições de baixa disponibilidade hídrica.
2. **Arquitetura Radicular Avançada:** Os seus sistemas radiculares profundos estendem-se por vários metros de profundidade, permitindo a exploração de horizontes inferiores do solo e o acesso a reservas de água subterrâneas não afetadas pela evaporação superficial imediata.

<br>

### 4.2. O Efeito de 'Lag' e a Memória Hidrológica do Solo

A introdução da variável desfasada no tempo `spi_12_lag1` revelou um fenómeno de extrema relevância para a compreensão da vulnerabilidade agrícola de longo prazo: a existência de uma "memória hidrológica" no ecossistema hídrico-planta. A modelação demonstrou que a produtividade de um determinado ano agrícola não é uma função exclusiva das condições meteorológicas que ocorrem durante o seu próprio ciclo de 12 meses.

Para as culturas lenhosas e arbóreas, o stress hídrico severo sofrido numa campanha manifesta impactos retardados que comprometem a produtividade da campanha subsequente, mesmo que esta última registe uma recuperação completa da pluviosidade ou seja classificada como um "Ano húmido". Este efeito de *lag* explica-se pelo facto de uma seca extrema esgotar as reservas hídricas profundas do solo e os lençóis freáticos, cuja taxa de recarga é lenta e cumulativa. Adicionalmente, o stress hídrico severo induz danos fisiológicos duradouros na estrutura das plantas perenes, tais como a redução do vigor das gemas florais formadas no ano anterior, a cavitação ou embolia nos vasos do xilema (que limita a capacidade futura de transporte de seiva) e o desvio de reservas de carbono da planta para mecanismos de sobrevivência e manutenção celular em detrimento da produção de fruto. Esta constatação sublinha o perigo de análises simplistas que consideram apenas os dados contemporâneos, ignorando o efeito cumulativo de secas plurianuais consecutivas.

<br>

### 4.3. Validação Estatística do Efeito 'Tampão' da Irrigação

Uma das contribuições mais robustas deste relatório reside na modelação inferencial executada através da regressão linear múltipla por Mínimos Quadrados Ordinários (OLS). A inclusão do **termo de interação** `spi_12:irrigable_area_k_ha` visou testar formalmente se a disponibilidade de infraestrutura de regada altera a sensibilidade da produtividade face à severidade da seca.

Os resultados do modelo revelaram que o coeficiente associado a este termo de interação é positivo e dotado de elevada significância estatística ($p < 0.05$). Este output matemático reveste-se de grande importância analítica: ele prova que a inclinação da reta que descreve a quebra de produtividade em função do decréscimo do SPI-12 é significativamente atenuada à medida que a área irrigável regional aumenta. A infraestrutura de regadio atua, por conseguinte, como um verdadeiro amortecedor hídrico ou efeito "tampão", quebrando a dependência linear e direta que vincula o sucesso das culturas à ocorrência imediata de precipitação atmosférica. A água fornecida artificialmente substitui o défice pluviométrico, mantendo o potencial de água no solo em níveis compatíveis com a atividade fotossintética e garantindo a estabilização dos rendimentos agrícolas.

Contudo, a interpretação destes resultados não pode isolar-se da dimensão da sustentabilidade ecológica e da gestão macro-hídrica. A conversão em grande escala de áreas agrícolas para sistemas intensivos de regadio (fenómeno impulsionado na região do Alentejo pelo Empreendimento Multifuncional de Alqueva) garante a resiliência económica imediata dos produtores face às secas, mas gera um dilema ecológico a longo prazo. O aumento contínuo da procura de água para fins agrícolas transfere o stress ambiental dos campos de cultivo para as bacias hidrográficas superficiais e para os aquíferos subterrâneos. Em cenários de secas severas generalizadas e plurianuais, a própria resiliência do sistema de regadio pode ser colocada em causa devido à redução drástica volumes armazenados nas barragens, acentuando os conflitos pelo uso da água entre os setores agrícola, urbano, industrial e ecológico.

<br>

### 4.4. Eficácia Metodológica do SPI-12 vs. Indicadores Absolutos

O modelo não-linear de Machine Learning implementado através do algoritmo *Random Forest Regressor* trouxe esclarecimentos metodológicos cruciais sobre a seleção de preditores em modelos agroclimáticos. Os perfis de importância de variáveis (*feature importances*) revelaram de forma consistente que o índice normalizado `spi_12` e a capacidade infraestrutural de regadio (`irrigable_area_ha`) possuem pesos explicativos significativamente superiores quando comparados com o volume bruto de precipitação anual acumulada (`precipitation_mm`).

Esta diferenciação matemática valida a escolha do SPI-12 como a métrica de referência para este tipo de estudo. Os indicadores de precipitação absoluta em milímetros são frequentemente falhos ao nível agregado porque mascaram a distribuição sazonal e temporal da água. Como evidenciado no Módulo II do projeto, uma região pode registar um total anual de milímetros considerado normal ou até mesmo acima da média devido a eventos de precipitação torrencial concentrados num curto período de inverno. No entanto, se os meses de primavera e verão coincidirem com uma ausência total de pluviosidade e temperaturas anormalmente elevadas, as culturas sofrerão um stress hídrico severo e quebras de rendimento devastadoras. O SPI-12, ao basear-se numa distribuição probabilística normalizada indexada aos desvios históricos de longo prazo, reflete com muito maior precisão a anomalia do teor de humidade no solo e o défice real disponível para o sistema radicular das plantas ao longo do ciclo de crescimento.

<br>

### 4.5. Implicações para Políticas Agrícolas e Estratégias de Adaptação

Os resultados integrados deste relatório convergem para a necessidade urgente de desenhar e implementar políticas públicas de planeamento e adaptação agrícola baseadas em evidências de dados para o Sul de Portugal, mitigando os riscos associados à transição climática em curso. As recomendações estratégicas dividem-se em três eixos principais:

1. **Modernização Tecnológica e Eficiência Hídrica Extrema:** Atendendo a que a irrigação provou ser o principal fator de mitigação de quebras agrícolas, os investimentos futuros não devem focar-se na expansão horizontal desmedida da área regada, mas sim na maximização da eficiência por gota de água utilizada. Isto implica incentivar a transição para sistemas de rega de precisão automatizados (como a rega gota-a-gota inteligente), geridos através de modelos preditivos baseados em sensores locais de humidade do solo e dados meteorológicos em tempo real, evitando perdas por evaporação e sobre-irrigação.

2. **Reconversão Estratégica de Culturas Vulneráveis:** O tecido agrícola do Alentejo e do Algarve deve adaptar-se de forma pró-ativa à inevitabilidade de regimes áridos e secas mais frequentes. Culturas anuais extensivas que demonstraram índices de vulnerabilidade insustentáveis devem ser progressivamente substituídas ou combinadas com sistemas de agricultura de conservação. Deve ser estimulada a expansão de culturas perenes com elevada resiliência comprovada (como variedades autóctones de Vinha e Olival adaptadas à secura) ou a introdução de novas espécies de elevado valor acrescentado e baixa exigência hídrica (como os frutos de casca rija, nomeadamente a Amendoeira e a Alfarrobeira, que se enquadram perfeitamente nos perfis de resiliência identificados pelo algoritmo de clustering).

3. **Promoção de Variedades Genéticas Resilientes:** Incentivar programas de investigação e desenvolvimento focados na seleção e introdução de genótipos agrícolas locais que exibam maior tolerância ao stress térmico e hídrico. A utilização de variedades com ciclos vegetativos curtos (capazes de escapar aos meses de maior canícula) ou dotadas de maior eficiência no uso da água permitirá salvaguardar a sustentabilidade económica das explorações agrícolas familiares e empresariais, reduzindo a pegada hídrica agregada e protegendo os ecossistemas do Sul do país.

## 5. Referências Bibliográficas

* **INE - Instituto Nacional de Estatística (2024).** *Estatísticas Agrícolas Anuais: Séries Temporais de Superfície, Produção e Rendimento das Principais Culturas por Região Agrária (2005-2018).* Lisboa, Portugal.

* **IPMA - Instituto Português do Mar e da Atmosfera (2024).** *Monitorização Climatológica e do Índice Padronizado de Precipitação (SPI-12) para as Estações de Beja e Faro.* Lisboa, Portugal.

* **McKee, T. B., Doesken, N. J., & Kleist, J. (1993).** *The relationship of drought frequency and duration to time scales.* In Proceedings of the 8th Conference on Applied Climatology (Vol. 17, pp. 179-183). Anaheim, California.